### PR TITLE
nanopb: 0.4.1 -> 0.4.3

### PR DESCRIPTION
**Pull request by lopsided98**
###### Motivation for this change

Updates nanopb to the latest version. Upstream changes allow us to remove most of the hacks from the package. All the `passthru.tests` succeed.

cc @kalbasit 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/nanopb/default.nix [hunk 1]
```diff
@@ -7,53 +7,37 @@
...
     repo = pname;
     rev = version;
-    sha256 = "16zxk42wzn519bpxf4578qn97k0h1cnbkvqqkqvka9sl0n3lz2dp";
+    sha256 = "sha256-32YU6LfPojvEVA2Z3DRZdGzDkSpAUgo1BueC/pl9qlY=";
   };
 
-  nativeBuildInputs = [ cmake python3 ];
+  nativeBuildInputs = [ cmake python3 python3.pkgs.wrapPython ];
+
+  pythonPath = with python3.pkgs; [ python3.pkgs.protobuf six ];
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/nanopb/default.nix:23]:
```suggestion
  pythonPath = with python3.pkgs; [ protobuf six ];
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 101988` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>nanopb</li>
  </ul>
</details>

File: pkgs/development/libraries/nanopb/default.nix (hunk 1 above)
Comment (lopsided98) [reply pkgs/development/libraries/nanopb/default.nix:23]: That doesn't work. `with` doesn't seem to override the protobuf from the outer scope.

Comment (SuperSandro2000) [reply pkgs/development/libraries/nanopb/default.nix:23]: Good catch. I did not check this.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 101988` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>nanopb</li>
  </ul>
</details>

**Pull request merged**